### PR TITLE
feat: revamped dag api

### PR DESCRIPTION
**Pull request by diasdavid**
ref: https://github.com/ipfs/interface-ipfs-core/issues/81#issuecomment-277271941

- [x] merge get and resolve
- [x] localResolve option
- [x] put with cid or with hashAlg and format
- [x] refactor dag-pb tests
- [x] new basic tests
- [x] refactor dag-cbor tests
- [x] refactor eth-block tests
- [x] refactor all-together-now tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,71 +14,68 @@
...
 ## Usage
 
 ```js
-const IPLDResolver = require('ipld-resolver')
+const Resolver = require('ipld-resolver')
 
 // pass an optional blockService, if no blockService is passed,
 // one is created in memory.
-const ipldResolver = new IPLDResolver(blockService)
+const Resolver = new Resolver(blockService)
```
Comment (nicola) [inline README.md]: let's tell them that the blockService can be optional

**Comment by diasdavid** [comment]
done, ready for review :)

File: README.md [hunk 2]
```diff
@@ -14,71 +14,68 @@
...
 ```
 
 ## API
 
-### IPLD Resolver
+### `.put(node, <<cid> || <format>, <hashAlg>>, callback)`
```
Comment (nicola) [inline README.md]: Wasn't it meant to be {node: *, cid: *} (this is what I saw from the tests)

File: src/index.js [hunk 3]
```diff
@@ -45,36 +45,51 @@ module.exports = class IPLDResolver {
...
+                     dagCBOR.resolver,
+                     dagCBOR.util)
+
+    this.support.add(ipldEthBlock.resolver.multicodec,
+                     ipldEthBlock.resolver,
+                     ipldEthBlock.util)
   }
 
-  resolve (cid, path, callback) {
+  get (cid, path, options, callback) {
```
Comment (nicola) [inline src/index.js]: <3 merging get and resolve :)

File: README.md (hunk 2 above)
Comment (diasdavid) [inline README.md]: see: https://github.com/ipfs/interface-ipfs-core/issues/81#issuecomment-277271941
tests were updated

Comment (nicola) [inline README.md]: I saw it later on in the code while I was seeing this.
This makes much more sense and it is much cleaner, awesome!

(it just breaks my ipld selector tests! :D)

File: README.md (hunk 1 above)
Comment (diasdavid) [inline README.md]: It's on the comment :)
image

File: README.md (hunk 2 above)
Comment (diasdavid) [inline README.md]: I see, sorry about that, as you say it gets things clearer :)

**Comment by kumavis** [comment]
nice!

File: README.md (hunk 1 above)
Comment (nicola) [inline README.md]: it was late night..

Comment (diasdavid) [inline README.md]: @nicola <3

**Pull request merged**